odoo/fields.py

```diff
@@ -989,6 +989,11 @@ def __get__(self, record, owner):
                         value = self.convert_to_cache(False, record, validate=False)
                         env.cache.set(record, self, value)
 
+            elif self.store and record._origin:
+                # new record with origin: fetch from origin
+                value = self.convert_to_cache(record._origin[self.name], record)
+                env.cache.set(record, self, value)
+
```

Comment:
this part is copy pasted after the `if self.compute:` to keep the minimal diff for display purpose.